python-sdk/src/astro/sql/operators/data_validations/ColumnCheckOperator.py

```diff
@@ -0,0 +1,242 @@
+    - leq_to: value that results should be less than or equal to
+    - tolerance: the percentage that the result may be off from the expected value
+
+    :param dataset: the table or dataframe to run checks on
+    :param column_mapping: the dictionary of columns and their associated checks, e.g.
+
+    .. code-block:: python
+
+        {
+            "col_name": {
+                "null_check": {
+                    "equal_to": 0,
+                },
+                "min": {
+                    "greater_than": 5,
+                    "leq_to": 10,
+                    "tolerance": 0.2,
+                },
+                "max": {"less_than": 1000, "geq_to": 10, "tolerance": 0.01},
+            }
+        }
+    """
+
+    def __init__(
+        self,
+        dataset: Union[BaseTable, pandas.DataFrame],
+        column_mapping: Dict[str, Dict[str, Any]],
+        partition_clause: Optional[str] = None,
+        task_id: Optional[str] = None,
+        **kwargs,
+    ):
+        for checks in column_mapping.values():
+            for check, check_values in checks.items():
+                self._column_mapping_validation(check, check_values)
+
+        self.dataset = dataset
+        self.column_mapping = column_mapping
+        self.partition_clause = partition_clause
+        self.kwargs = kwargs
+        self.df = None
+
+        dataset_qualified_name = ""
+        dataset_conn_id = ""
+
+        if isinstance(dataset, BaseTable):
+            db = create_database(conn_id=self.dataset.conn_id)  # type: ignore
+            self.conn_id = self.dataset.conn_id
+            dataset_qualified_name = db.get_table_qualified_name(table=self.dataset)
+            dataset_conn_id = dataset.conn_id
+
+        super().__init__(
+            table=dataset_qualified_name,
+            column_mapping=self.column_mapping,
+            partition_clause=self.partition_clause,
+            conn_id=dataset_conn_id,
+            task_id=task_id if task_id is not None else get_unique_task_id("column_check"),
+        )
+
+    def execute(self, context: "Context"):
+        if isinstance(self.dataset, BaseTable):
+            return super().execute(context=context)
+        elif type(self.dataset) == pandas.DataFrame:
+            self.df = self.dataset
+        else:
+            raise ValueError("dataset can only be of type pandas.dataframe | Table object")
+
+        self.process_checks()
+
+    def get_check_result(self, check_name: str, column_name: str, df: pandas.DataFrame):
+        """
+        Get the check method results post validating the dataframe
+        """
+        if df is not None and column_name in df.columns:
+            column_checks = {
+                "null_check": self.col_null_check,
+                "distinct_check": self.col_distinct_check,
+                "unique_check": self.col_unique_check,
+                "min": self.col_min,
+                "max": self.col_max,
```

Comment:
```suggestion
                "null_check": lambda col: col.isna().sum(),
                "distinct_check": self.col_distinct_check,
                "unique_check": self.col_unique_check,
                "min": self.col_min,
                "max": self.col_max,
```
Maybe it makes sense to either use lambdas as the user provides a name (of the function) already, or we directly use `getattr` to resolve the name into the function. That way you don't require an explicit mapping.

Reply:
Done.